### ISSUE TITLE
error when populating a self referencing collection with a custom junction table

### ISSUE THREAD

**Issue by jojo897** [balderdashy/sails#5690]
**Waterline version**: 0.11.6
**Node version**: 4.2.6
**NPM version**: 3.10.8
**Operating system**: ubuntu LTS 16.04

<hr/>

**1st scenario:**
**User**

    
    autoPK: false,
    autoCreatedAt: false,
    autoUpdatedAt: false,
    tableName: "User",
    attributes: {
        userId: {
            primaryKey: true,
            type: "integer",
            autoIncrement: true,
            unique: true
        },
        friends: {
        	collection: "user",
        	via: "receiver",
        	through: "friendship"
        },
        friendsof: {
        	collection: "user",
        	via:"sender",
        	through: "friendship"
        }
    }
    

 **Friendship**

    autoPK: false,
    autoCreatedAt: false,
    autoUpdatedAt: false,
    tableName: "Friendship",
    junctionTable: true,

    attributes: {
        receiver: {
            type: "integer",
            columnName: "receiver",
            foreignKey: true,
            references: "user",
            on: 'userId',
            via: 'sender'
        },
        sender: {
            type: "integer",
            columnName: "sender",
            foreignKey: true,
            references: "user",
            on: "userId",
            via: 'receiver'
        }
    }
    
Trying to populate "friends" will work but the result data will not be as expected.
Instead of having a collection of random users as friends, you'll get a collection of yourself as friend.
However populating "friendsof" will work and the result will be correct.

**2nd Scenario:**
 **Friendship**

    autoPK: false,
    autoCreatedAt: false,
    autoUpdatedAt: false,
    tableName: "Friendship",
    junctionTable: true,

    attributes: {
        sender: {
            type: "integer",
            columnName: "sender",
            foreignKey: true,
            references: "user",
            on: "userId",
            via: 'receiver'
        },
        receiver: {
            type: "integer",
            columnName: "receiver",
            foreignKey: true,
            references: "user",
            on: 'userId',
            via: 'sender'
        }
    }
    
If you declare sender before receiver like this:
Populating "friends" will work correctly whereas populating "friendsof" will fail in the same way as described in scenario 1.

So basically it's a matter of finding the right attribute to join on when populating.
